apps/client/sttr_build/parser.js

```diff
@@ -52,165 +75,288 @@ const getAutofillResolverKey = (questionText) => {
   }
 };
 
+const filt = (arr, tagName) => arr.filter(({ name }) => name === tagName);
+
+const find = (arr, tagName) => arr.find(({ name }) => name === tagName);
+
+const log = (obj) => {
+  if (obj === undefined) return "undefined";
+  if (obj.hasOwnProperty("length") && typeof obj === "object") {
+    // assume collection
+    return obj.map(log);
+  } else {
+    const { attribs, children, data, name, type } = obj;
+    return type === "text"
+      ? {
+          data,
+        }
+      : {
+          attribs,
+          children: children && children.map(log),
+          name,
+        };
+  }
+};
+
 /**
- * A parser for STTR-XML files
+ * Get decisions configuration
+ *
+ * @returns {any} a configuration object for decisions
  */
-class Parser {
-  /**
-   * @param {string} xml - STTR-XML
-   */
-  constructor(xml) {
-    const parsed = parser.parse(xml, {
-      ignoreAttributes: false,
-      arrayMode: true,
-    });
+// function getDecisions(xmlDecisions: XMLDecisionType[]) {
+function getDecisions(xmlDecisions) {
+  debug("xmlDecisions", log(xmlDecisions));
```

Comment:
I see you created a debug function thats logs information if logging is enabled. But by just seeing debug you would think this shouldn't be pushed or something. Maybe we need a better name

Reply:
I'm following the pattern from this package: https://www.npmjs.com/package/debug
It maybe needs a bit of getting used to, but a lot of big packages use this package.